src/wallet/wallet.cpp

```diff
@@ -1929,16 +1929,7 @@ bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount &nFeeRet, int& nC
     // Add new txins (keeping original txin scriptSig/order)
     BOOST_FOREACH(const CTxIn& txin, wtx.vin)
     {
-        bool found = false;
-        BOOST_FOREACH(const CTxIn& origTxIn, tx.vin)
-        {
-            if (txin.prevout.hash == origTxIn.prevout.hash && txin.prevout.n == origTxIn.prevout.n)
-            {
-                found = true;
-                break;
-            }
-        }
-        if (!found)
+        if (!coinControl.IsSelected(txin.prevout))
```

Comment:
Yes they are. The only difference is that the old code performs a linear search and reimplements the `COutPoint::operator==` inline.

`coinControl.setSelected` is filled with the current inputs, which are the ones to search when the inputs are added, otherwise the same input would be duplicated.